examples/computecloth/computecloth.cpp

```diff
@@ -306,26 +300,66 @@ class VulkanExample : public VulkanExampleBase
 
 			// Dispatch the compute job
 			const uint32_t iterations = 64;
+			static_assert(iterations % 2 == 0, "The below code assumes an even number of iterations.");
+			uint32_t readSet{ 0 };
 			for (uint32_t j = 0; j < iterations; j++) {
-				readSet = 1 - readSet;
+				bool lastIteration = j == iterations - 1;
 				vkCmdBindDescriptorSets(compute.commandBuffers[i], VK_PIPELINE_BIND_POINT_COMPUTE, compute.pipelineLayout, 0, 1, &compute.descriptorSets[readSet], 0, 0);
 
-				if (j == iterations - 1) {
+				if (lastIteration) {
 					calculateNormals = 1;
 					vkCmdPushConstants(compute.commandBuffers[i], compute.pipelineLayout, VK_SHADER_STAGE_COMPUTE_BIT, 0, sizeof(uint32_t), &calculateNormals);
 				}
 
 				vkCmdDispatch(compute.commandBuffers[i], cloth.gridsize.x / 10, cloth.gridsize.y / 10, 1);
 
-				// Don't add a barrier on the last iteration of the loop, since we'll have an explicit release to the graphics queue
-				if (j != iterations - 1) {
-					addComputeToComputeBarriers(compute.commandBuffers[i]);
+				// Use a barrier to ensure that writes are finished before the next dispatch
+				if (!lastIteration) {
+					addComputeToComputeBarriers(compute.commandBuffers[i], readSet != 0);
 				}
-
+				readSet = 1 - readSet;
 			}
 
-			// release the storage buffers back to the graphics queue
-			addComputeToGraphicsBarriers(compute.commandBuffers[i], VK_ACCESS_SHADER_WRITE_BIT, 0, VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT, VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT);
+			// The last iteration wrote to the "input" buffer, so we need copy the results to the output buffer
+			// First we need to put the buffers into transfer mode
+			std::array<VkBufferMemoryBarrier, 2> bufferBarriers;
```

Comment:
Because the last dispatch is odd, it wrote to `input`, so we want to copy to `output`, so we need a barrier for the copy